cmd/kubeadm/app/cmd/phases/markmaster.go

```diff
@@ -25,68 +25,65 @@ import (
-		# Applies master label and taint to the current node, functionally equivalent to what executed by kubeadm init.
-		kubeadm alpha phase mark-master
+	
+	markcontrolplaneExample = normalizer.Examples(`
+		# Run master markcontrolplane checks using a config file.
+		kubeadm init phase mark-controlplane --config config.yml
 
 		# Applies master label and taint to a specific node
-		kubeadm alpha phase mark-master --node-name myNode
+		kubeadm init phase mark-controlplane --node-name myNode
 		`)
 )
 
-// NewCmdMarkMaster returns the Cobra command for running the mark-master phase
-func NewCmdMarkMaster() *cobra.Command {
-	cfg := &kubeadmapiv1beta1.InitConfiguration{}
-
-	// Default values for the cobra help text
-	kubeadmscheme.Scheme.Default(cfg)
-
-	var cfgPath string
-	kubeConfigFile := kubeadmconstants.GetAdminKubeConfigPath()
 
-	cmd := &cobra.Command{
-		Use:     "mark-master",
-		Short:   "Mark a node as master",
-		Long:    markMasterLongDesc,
-		Example: markMasterExample,
-		Aliases: []string{"markmaster"},
-		Run: func(cmd *cobra.Command, args []string) {
-			if err := validation.ValidateMixedArguments(cmd.Flags()); err != nil {
-				kubeadmutil.CheckErr(err)
-			}
-
-			kubeConfigFile = cmdutil.FindExistingKubeConfig(kubeConfigFile)
-			client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
-			kubeadmutil.CheckErr(err)
+type markcontrolplaneData interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	DryRun() bool
+}
 
-			// KubernetesVersion is not used, but we set it explicitly to avoid the lookup
-			// of the version from the internet when executing ConfigFileAndDefaultsToInternalConfig
-			SetKubernetesVersion(cfg)
+// NewmarkcontrolplanePhase creates a kubeadm workflow phase that implements mark-controlplane checks.
+func NewmarkcontrolplanePhase() workflow.Phase {
+	return workflow.Phase{
+		Name:    "markcontrolplane",
+		Short:   "Run master markcontrolplane checks",
+		Long:    "Run master markcontrolplane checks, applies master label.",
+		Example: markcontrolplaneExample,
+		Run:     runmarkcontrolplane,
+	}
+}
 
-			// This call returns the ready-to-use configuration based on the configuration file that might or might not exist and the default cfg populated by flags
-			internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, cfg)
-			kubeadmutil.CheckErr(err)
+// runmarkcontrolplane executes markcontrolplane checks logic.
+func runmarkcontrolplane(c workflow.RunData) error {
+	data, ok := c.(markcontrolplaneData)
+	if !ok {
+		return fmt.Errorf("markcontrolplane phase invoked with an invalid data struct")
+	}
 
-			err = markmasterphase.MarkMaster(client, internalcfg.NodeRegistration.Name, internalcfg.NodeRegistration.Taints)
-			kubeadmutil.CheckErr(err)
-		},
+	fmt.Println("[markcontrolplane] running markcontrolplane checks")
+	var cfgPath string
+    kubeConfigFile := kubeadmconstants.GetAdminKubeConfigPath()
+	kubeConfigFile = cmdutil.FindExistingKubeConfig(kubeConfigFile)
```

Comment:
note to self: seems to me like we are not doing this in other places, where we should be doing it.
possibly best to move it in init.go